hack/boilerplate/boilerplate.py

```diff
@@ -181,16 +185,95 @@ def get_regexs():
     regexs["go_build_constraints"] = re.compile(r"^(// \+build.*\n)+\n", re.MULTILINE)
     # strip #!.* from shell scripts
     regexs["shebang"] = re.compile(r"^(#!.*\n)\n*", re.MULTILINE)
+    # Search for generated files
+    regexs["generated"] = re.compile( 'DO NOT EDIT' )
     return regexs
 
+
+def is_generated_files(filename, regexs):
+    for d in skipped_ungenerated_files:
+ 	if d in filename:
+            return False
+    
+    try:
+        f = open(filename, 'r')
+    except Exception as exc:
+        print("Unable to open %s: %s" % (filename, exc), file=verbose_out)
+        return False
+
+    data = f.read()
+    f.close()
+
+    p = regexs["generated"]
+    found = p.search(data)
+
+    if found:
+        return True
+    else:
+        return False
+
+
+def generated_file_passes(filename, refs, regexs):
+    try:
+        f = open(filename, 'r')
+    except Exception as exc:
+        print("Unable to open %s: %s" % (filename, exc), file=verbose_out)
+        return False
+
+    data = f.read()
+    f.close()
+
+    extension = "generatego"
+    ref = refs[extension]
+
+    # remove build tags from the top of Go files
+    p = regexs["go_build_constraints"]
+    (data, found) = p.subn("", data, 1)
+
+    data = data.splitlines()
+
+    # if our test file is smaller than the reference it surely fails!
+    if len(ref) > len(data):
+        print('File %s smaller than reference (%d < %d)' %
```

Comment:
does this really need a print?

Reply:
It's just a copy from func #[file_passes](https://github.com/kubernetes/kubernetes/blob/master/hack/boilerplate/boilerplate.py#L96)

Reply:
yeah, it has some duplicated code with #file_passes, but also has some difference  in the middle. :(
>does this really need a print?

in the func print, has flag verbose like: print("Unable to open %s: %s" % (filename, exc), file=verbose_out) 
just used for debug, we can run like:
~#: python hack/boilerplate/boilerplate.py -v
then will print the debug log.

Reply:
I think it's fine. It looked like leftover debug output, but it isn't.